### ISSUE TITLE
Strange behavior of spark schema generator for Decimal type

### ISSUE THREAD

**Issue by yarik3571** [mitchelllisle/sparkdantic#356]
I am running `sparkdantic==0.23.0`
And different spark schema is generated between calls to `model_spark_schema()`
E.g.
```
from decimal import Decimal

from pydantic import Field
from sparkdantic import SparkModel


class DecimalModel(SparkModel):
    a: Decimal
    b: Decimal = Field(decimal_places=2)
    c: Decimal = Field(decimal_places=2, max_digits=5)


if __name__ == '__main__':
    print(DecimalModel.model_spark_schema())
    print(DecimalModel.model_spark_schema())
    print(DecimalModel.model_spark_schema())
```
Outputs:
```
StructType([StructField('a', DecimalType(10,0), False), StructField('b', DecimalType(10,2), False), StructField('c', DecimalType(5,2), False)])
StructType([StructField('a', DecimalType(10,0), False), StructField('b', DecimalType(10,0), False), StructField('c', DecimalType(10,0), False)])
StructType([StructField('a', DecimalType(10,0), False), StructField('b', DecimalType(10,0), False), StructField('c', DecimalType(10,0), False)])
```
So, decimal_places and max_digits are being ignored in second and third calls.

Possible reason is that [here ](https://github.com/mitchelllisle/sparkdantic/pull/193/files#diff-d0724184b9d94a5ae82a0fe7de0cbd5e7968b40f413fff766b849be2b44d8f62R330) `pop()` from `metadata` is done that actually removes element from `metadata` itself

**Comment by dan1elt0m** [comment]
PR with fix: https://github.com/mitchelllisle/sparkdantic/pull/357

**Comment by mitchelllisle** [comment]
Released in [v0.24.0](https://github.com/mitchelllisle/sparkdantic/releases/tag/v0.24.0)
